frontend/packages/dev-console/src/components/topology/TopologyDataRenderer.tsx

```diff
@@ -1,37 +1,38 @@
 import * as React from 'react';
+import { useParams } from 'react-router-dom';
 import { Model } from '@patternfly/react-topology';
 import { TopologyDataResources, TrafficData } from './topology-types';
 import ModelContext, { ExtensibleModel } from './data-transforms/ModelContext';
 import { baseDataModelGetter } from './data-transforms';
 
 export interface RenderProps {
   model?: Model;
-  namespace: string;
   loaded: boolean;
   loadError: string;
+  namespace: string;
 }
 
 export interface TopologyDataRendererProps {
   kindsInFlight: boolean;
   resources: TopologyDataResources;
   render(props: RenderProps): React.ReactElement;
-  namespace: string;
   trafficData?: TrafficData;
 }
 
 export const TopologyDataRenderer: React.FC<TopologyDataRendererProps> = ({
   render,
   resources,
-  namespace,
   kindsInFlight,
   trafficData,
 }) => {
   const dataModelContext = React.useContext<ExtensibleModel>(ModelContext);
   const [model, setModel] = React.useState<Model>(null);
   const [loadError, setLoadError] = React.useState<string>(null);
+  const params = useParams();
```

Comment:
I think we are better off to pass in props than couple the component to URL params. Especially when the component is utility in nature.